packages/common/src/components/CopyableToClipboard/index.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import Clipboard from 'clipboard';
+import { invokeGlobalErrorHandler } from '../../utilities/splash.screen';
+
+interface IProps {
+  textGetter: () => string;
+  globallyUniqueSelector?: string;
+  onSuccess?: () => void;
+  onError?: () => void;
+}
+
+interface IState {}
+
+class CopyableToClipboard extends React.Component<IProps, IState> {
+  private idIfAny: string;
+  private clipboard: Clipboard;
+
+  constructor(props: IProps) {
+    super(props);
+
+    if (!this.props.globallyUniqueSelector) {
+      this.idIfAny = 'export-to-clipboard-' + CopyableToClipboard.nextGlobalId++;
+    }
+  }
+
+  componentDidMount() {
+    const selector = this.props.globallyUniqueSelector || `#${this.idIfAny} > *`;
+    this.clipboard = new Clipboard(selector, {
+      text: this.props.textGetter,
+    });
+
+    if (this.props.onSuccess) {
+      this.clipboard.on('success', this.props.onSuccess);
+    }
+
+    this.clipboard.on(
+      'error',
+      this.props.onError ? this.props.onError : invokeGlobalErrorHandler,
```

Comment:
might be too aggressive, don't you think?

Reply:
I can't think of any reason why clipboard would fail, ever.  And I've never seen it happen in practice.
As such, it feels like if it does fail, we will want to know.  And that way too the user doesn't think that they copied something when in fact they hadn't.

Reply:
that makes sense, but don't you think a console error could suffice?